compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -556,7 +557,11 @@ fn struct_def_set_fields(
 
                 match name_tokens.first() {
                     Some(Token::Ident(name)) if name_tokens.len() == 1 => {
-                        Ok((Ident::new(name.clone(), field_location.span), typ))
+                        Ok(hir_def::types::StructField {
+                            visibility: ItemVisibility::Private,
```

Comment:
Why not default to public members or else accept visibility parameters for each field, e.g. `[(Quoted, Type)] -> [(Quoted, Type, ItemVisibility)]` or similar?

Reply:
Ah, regarding these built-in macro methods, I didn't want to change them because it would be a breaking change and it'll break macros on the Aztec side. But maybe that's fine? We'd have to think of an API to define the visibility, and possibly also return the visibility when asking for struct fields. But maybe it's better to do that change in a separate PR? What do you think?

Reply:
I guess for now we could default to public to avoid this macro usage from introducing warnings. I'll do that :-)

Reply:
>  it would be a breaking change and it'll break macros on the Aztec side.

To avoid breakages, we could define a separate function `set_fields_with_visibility`
(And `StructDefinition::fields_with_visibility`)

> But maybe it's better to do that change in a separate PR? What do you think?

Sure, please add an issue for that

Reply:
Done: #6229